domains/client/consensus-relay-chain/Cargo.toml

```diff
@@ -8,6 +8,7 @@ edition = "2021"
 [dependencies]
 async-trait = "0.1.57"
 sc-consensus = { git = "https://github.com/subspace/substrate", rev = "66d3f73c5ef66c975fa5f54b6736ccd6821e395f" }
+sc-consensus-subspace = { version = "0.1.0", path = "../../../crates/sc-consensus-subspace" }
```

Comment:
Folks, this very not good.

You're making it impossible to compile things in parallel with these changes and pull things that this crate shouldn't be aware of. We already had subscriptions about notifications in other places and it was never necessary to expose internals of `sc-consensus-subspace`. Please revert this change and changes to `sc-consensus-subspace` and replace with proper abstractions. If there were other similar changes done recently, they need to be reverted and replaced with proper abstractions instead.

I understand it is easier to just make everything public and make everything depend on everything, but it hurts maintainability A LOT.

Reply:
Would be great if we can add some dependency rules in the contributing guide so that such things won't be missed easily, which may also be enforced in CI someday when there is more manpower on the devops team.

Reply:
Sure. I made the changes to get the access to notification stream easily. Will change it back soon 👍🏼